.github/workflows/ci.yaml

```diff
@@ -0,0 +1,81 @@
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches:
+    - main
+    - track/**
+
+# kill all previous jobs in the same branch if new commits are pushed
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  quality-checks:
+    name: Quality Checks
+    strategy:
+      matrix:
+        charm:
+          - kfp-api
+          - kfp-persistence
+          - kfp-profile-controller
+          - kfp-schedwf
+          - kfp-ui
+          - kfp-viewer
+          - kfp-viz
+    uses: canonical/charmed-kubeflow-workflows/.github/workflows/_quality-checks.yaml@all
+    with:
+      charm-path: ./charms/${{ matrix.charm }}
+      charm-name: ${{ matrix.charm }}
+
+  unit-tests:
+    name: Unit Tests
+    strategy:
+      matrix:
+        charm:
+          - kfp-api
+          - kfp-persistence
+          - kfp-profile-controller
+          - kfp-schedwf
+          - kfp-ui
+          - kfp-viewer
+          - kfp-viz
+    uses: canonical/charmed-kubeflow-workflows/.github/workflows/_unit-tests.yaml@all
+    secrets: inherit
+    with:
+      charm-name: ${{ matrix.charm }}
+
+  integration-tests:
+    name: Integration Tests
+    needs: quality-checks
+    strategy:
+      matrix:
+        charm:
+          - kfp-api
+          - kfp-profile-controller
+    uses: canonical/charmed-kubeflow-workflows/.github/workflows/_integration-tests.yaml@all
+    secrets: inherit
+    with:
+      charm-name: ${{ matrix.charm }}
+
+  bundle-tests:
+    name: Bundle Tests
+    needs: quality-checks
```

Comment:
I don't love the name "bundle tests", though we have a `bundle.yaml` file, I don't think we promote this as a bundle. I'd rather name this "Repository Level E2E Tests" or similar, to avoid confusion with:
1) the CKF bundle
2) thinking we have a pipelines bundle